### PR TITLE
First draft for API Search

### PR DESCRIPTION
**Pull request by fermujica**

### CODE REVIEW

File: modules/ROOT/pages/api-search.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Because Anypoint Studio 7.4 enables you to import your API specifications from Exchange or Maven as dependencies of your project, those dependencies no longer locate as resources inside the `src/resources/api/` directory, but as ZIP file dependencies in your project's classpath. This change makes your API dependencies not searchable through the File Search option in Studio. To enable you to search for specific text inside your API dependencies, Studio introduces API Search.
+
+You can access the API Search menu from the taskbar at the top of the Anypoint Studio display, selecting *Search* > *Search...*, or by using the `control + H` keyboard shortcut.
+
+image::api-search.png[]
+
+. Type the text that you want to search inside your API dependencies. +
+* Use `\*` to search for any string. For example, `*-api-*` searches for strings containing `-api-`.
+* Use `?` to allow any character after a specific pattern. For example, `customer?` searches for both `customer` and `customers`.
+* Use `\` to escape any of the reserved literals used in search patterns. For example, if you want to search for strings containing `\*`, `?`, or `\` you must escape them: `\*`, `\?`, and `\\`.
```
Comment (antonelladellanzo) [inline modules/ROOT/pages/api-search.adoc]: Here, when you say 

> For example, if you want to search for strings containing `\*`, `?`, or `\`

That \\* should be *

File: modules/ROOT/pages/api-search.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+. If you want to specify a specific name patter for the files you want to search through, type the pattern in the *File name patterns* field.
```
Comment (antonelladellanzo) [inline modules/ROOT/pages/api-search.adoc]: You've missed the `n` in `pattern` (it says `patter`)

File: modules/ROOT/pages/api-search.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
++
+* Use `\*` to search for any string. For example, `*-api-*` searches inside any API specification containing `-api-` in its name.
+* Use `?` to allow any character after a specific pattern. For example, `customer?.raml` searches inside both `customer.raml` and `customers.raml`.
+* Use `!` to exclude certain filenames or patterns. For example, `!examples.raml` searches inside all files except `examples.raml`.
+. Click the *Choose...* button to filter per file type.
+. Select the type of API specification in your project.
++
+* *API dependencies* searches inside API specifications imported from Exchange or Maven.
+* *Local API Files* searches inside API specifications added from a local file, or Design Center.
```
Comment (antonelladellanzo) [inline modules/ROOT/pages/api-search.adoc]: That comma after file seems weird. We should remove it or added it in the API Dependencies explanation to be consistent.

File: modules/ROOT/pages/api-search.adoc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+When you click *Search* Studio searches for your text criteria inside your API specifications.
+
+You can also search for all API specifications inside your open projects by selecting *Navigate* > *Open API Resource* in the taskbar at the top of the Anypoint Studio display, or by using the `cmd + Shift P` (OSx) or `control + shift + P` (Windows/Linux) keyboard shortcut.
```
Comment (antonelladellanzo) [inline modules/ROOT/pages/api-search.adoc]: When you say the command for OSx, you've missed the + before de P

**Review by antonelladellanzo — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/api-search.adoc (hunk 1 above)
Comment (fermujica) [reply modules/ROOT/pages/api-search.adoc]: Yes, that's to escape the asciidoc reserved character `*` . 
It renders without the `\` in HTML 👍

Comment (antonelladellanzo) [reply modules/ROOT/pages/api-search.adoc]: Ohh perfect. But then, after that when you say:
> you must escape them: `\*`

That will be shown as only `*`?? or `\*`?

File: modules/ROOT/pages/api-search.adoc [hunk 5]
```diff
@@ -0,0 +1,38 @@
+= Search Inside Your API Specification Dependencies
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Because Anypoint Studio 7.4 enables you to import your API specifications from Exchange or Maven as dependencies of your project, those dependencies no longer locate as resources inside the `src/resources/api/` directory, but as ZIP file dependencies in your project's classpath. This change makes your API dependencies not searchable through the File Search option in Studio. To enable you to search for specific text inside your API dependencies, Studio introduces API Search.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
When you use Anypoint Studio 7.4 to import your API specifications from Exchange or Maven as dependencies of your project, those dependencies relocate from`src/resources/api/` to a ZIP file in your project's classpath. This change means that you cannot search for your API dependencies by using the File Search option in Studio. To enable you to search for specific text inside your API dependencies, Studio introduces the API Search feature.
```

File: modules/ROOT/pages/api-search.adoc [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+You can access the API Search menu from the taskbar at the top of the Anypoint Studio display, selecting *Search* > *Search...*, or by using the `control + H` keyboard shortcut.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
You can access the API Search menu from the taskbar at the top of the Anypoint Studio display, either by selecting *Search* > *Search...* or by using the control and H keys on your keyboard simultaneously:
```
I was tempted to suggest "CTRL+H", but I realized that that might be too Windows-centric....

File: modules/ROOT/pages/api-search.adoc [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+image::api-search.png[]
+
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc:11]: To search inside of your API specification dependencies:

File: modules/ROOT/pages/api-search.adoc [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+. Type the text that you want to search inside your API dependencies. +
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
. Type the text that you want to search for, following these conventions: +
```

File: modules/ROOT/pages/api-search.adoc [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+* Use `\*` to search for any string. For example, `*-api-*` searches for strings containing `-api-`.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]: GLOBALLY limit list items and steps to one sentence or fragment, moving related info directly below:
```suggestion
* Use `\*` to search for any string. 
For example, `*-api-*` searches for strings containing `-api-`.
```

File: modules/ROOT/pages/api-search.adoc [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+* Use `?` to allow any character after a specific pattern. For example, `customer?` searches for both `customer` and `customers`.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
* Append `?` to your search string to allow any character at the end of the string. 
For example, `customer?` searches for both `customer` and `customers`.
```

Comment (kbostrom) [reply modules/ROOT/pages/api-search.adoc]: Would it also search for `customer-related`? Just curious as to the limits of what or how many characters can be appended.

File: modules/ROOT/pages/api-search.adoc (hunk 1 above)
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
* Use `\` to escape any of the reserved literals used in search patterns. 
For example, if you want to search for strings containing `\*`, `?`, or `\` you must escape them: `\*`, `\?`, and `\\`.
```

File: modules/ROOT/pages/api-search.adoc [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+* Use `\` to escape any of the reserved literals used in search patterns. For example, if you want to search for strings containing `\*`, `?`, or `\` you must escape them: `\*`, `\?`, and `\\`.
+. If you want to specify a specific name pattern for the files you want to search through, type the pattern in the *File name patterns* field.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
. If you want to specify a specific name pattern for the files you want to search through, type the pattern in the *File name patterns* field, following these conventions:
```
OR
```suggestion
. If you want to specify a specific name pattern for the files you want to search through, type the pattern in the *File name patterns* field, following the previously described conventions plus this one:
```
...followed only by the additional convention description.

File: modules/ROOT/pages/api-search.adoc [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
++
+* Use `\*` to search for any string. For example, `*-api-*` searches inside any API specification containing `-api-` in its name.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]: ditto prev edit to same text

File: modules/ROOT/pages/api-search.adoc [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+* Use `?` to allow any character after a specific pattern. For example, `customer?.raml` searches inside both `customer.raml` and `customers.raml`.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]: ditto

File: modules/ROOT/pages/api-search.adoc [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+* Use `!` to exclude certain filenames or patterns. For example, `!examples.raml` searches inside all files except `examples.raml`.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
* Use `!` to exclude certain filenames or patterns. 
For example, `!examples.raml` searches inside all files except `examples.raml`.
```

File: modules/ROOT/pages/api-search.adoc [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+. Click the *Choose...* button to filter per file type.
+. Select the type of API specification in your project.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
. Select one of the following types of API specifications in your project:
```

File: modules/ROOT/pages/api-search.adoc [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
++
+* *API dependencies* searches inside API specifications imported from Exchange or Maven.
+* *Local API Files* searches inside API specifications added from a local file or Design Center.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
* *Local API Files* searches inside API specifications added from a local file or from Anypoint Design Center.
```

File: modules/ROOT/pages/api-search.adoc [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+
+When you click *Search* Studio searches for your text criteria inside your API specifications.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
. Click *Search*. 
Studio searches for your text criteria inside your API specifications.
```

File: modules/ROOT/pages/api-search.adoc [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+
+You can also search for all API specifications inside your open projects by selecting *Navigate* > *Open API Resource* in the taskbar at the top of the Anypoint Studio display, or by using the `cmd + Shift + P` (OSx) or `control + shift + P` (Windows/Linux) keyboard shortcut.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
You can also search for all API specifications inside your open projects either by selecting *Navigate* > *Open API Resource* in the taskbar at the top of the Anypoint Studio display, or by using `cmd+Shift+P` (OSx) or `control+shift+P` (Windows/Linux) keyboard shortcuts:
```

File: modules/ROOT/pages/api-search.adoc [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+
+image::open-api-resource.png[]
+
+. Type the name of the API resource that you are looking for.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
. Type the name of the API resource that you are looking for:
```

File: modules/ROOT/pages/api-search.adoc [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+
+image::open-api-resource-result.png[]
+
+Studio displays all possible matches for the API resource that you typed. Select the resource you were looking for and click *OK*.
```
Comment (kbostrom) [inline modules/ROOT/pages/api-search.adoc]:
```suggestion
Studio displays all possible matches for the API resource that you typed. 
. Select the resource you were looking for and click *OK*.
```

**Review by kbostrom — APPROVED** [review]
Here you go!

**Review by antonelladellanzo — APPROVED** [review]

**Pull request merged**